### ISSUE TITLE
[Bug] Phone Auth timeout is being called immediately after code sent even though phoneAuthTimeoutMs is set to 60 * 1000

### ISSUE THREAD

**Issue by adityathoutam** [firebase/quickstart-unity#968]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2020.2.6f1
  * Firebase Unity SDK version: 7.1.0
  * Source you installed the SDK: UPM
  * Problematic Firebase Component: Auth
  * Other Firebase Components in use: Auth
  * Additional SDKs you are using: Nothing 
  * Platform you are using the Unity editor on: Mac
  * Platform you are targeting: Android 
  * Scripting Runtime: IL2CPP

### [REQUIRED] Please describe the issue here:

(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)
Phone Auth is not waiting for code auto retrieval, timeout is being called almost immediately even if timeout is set to 60 * 1000

2021-02-25 13:29:13.670 9331 19612 Info Unity Phone Auth, code sent
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.Sample.Auth.UIHandler:DebugLog(String)
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.Auth.CodeSent:Invoke(String, ForceResendingToken)
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.Auth.<CodeSentHandler>c__AnonStorey2:<>m__0()
2021-02-25 13:29:13.670 9331 19612 Info Unity System.Action:Invoke()
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.ExceptionAggregator:Wrap(Action)
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.AppUtilPINVOKE:PollCallbacks()
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.AppUtil:PollCallbacks()
2021-02-25 13:29:13.670 9331 19612 Info Unity Firebase.Platform.FirebaseHandler:Update()
2021-02-25 13:29:13.670 9331 19612 Info Unity 
2021-02-25 13:29:14.939 9331 20322 Warn FirebaseAuth [SmsRetrieverHelper] Timed out waiting for SMS.
2021-02-25 13:29:14.950 9331 19612 Info Unity Phone Auth, auto-verification timed out
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.Sample.Auth.UIHandler:DebugLog(String)
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.Auth.CodeAutoRetrievalTimeOut:Invoke(String)
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.Auth.<TimeOutHandler>c__AnonStorey3:<>m__0()
2021-02-25 13:29:14.950 9331 19612 Info Unity System.Action:Invoke()
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.ExceptionAggregator:Wrap(Action)
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.AppUtilPINVOKE:PollCallbacks()
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.AppUtil:PollCallbacks()
2021-02-25 13:29:14.950 9331 19612 Info Unity Firebase.Platform.FirebaseHandler:Update()
2021-02-25 13:29:14.950 9331 19612 Info Unity 

#### Steps to reproduce:

Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?
Yes
What's the issue repro rate? (eg 100%, 1/5 etc)
100%
What happened? How can we make the problem occur?
Check for Phone Auth and see that timeout is ignored everytime

This could be a description, log/console output, etc.

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by alexames** [comment]
I just want to make sure: when you ran the sample app, did you first follow the [directions in the readme file](https://github.com/firebase/quickstart-unity/tree/master/auth/testapp#android) including the [directions for running the sample](https://github.com/firebase/quickstart-unity/tree/master/auth/testapp#using-the-sample)? If the console is not properly configured to handle phone auth that could cause it to fail.

**Comment by adityathoutam** [comment]
yes I have read and made sure to check with quickstart app and our production app as well which uses Play Store signed SHA keys.

**Comment by chkuang-g** [comment]
Just to clarify, did you enable `Phone` as sign-in method on your Firebase console?

I was trying quickstart and I did notice that it timed out immediately as well.  I'm not sure what changed.  Will need to consult with the team.

**Comment by adityathoutam** [comment]
> Just to clarify, did you enable `Phone` as sign-in method on your Firebase console?
> 
> I was trying quickstart and I did notice that it timed out immediately as well. I'm not sure what changed. Will need to consult with the team.

Yes Phone is enabled and users are able to enter OTP normally and sign in.

**Comment by chkuang-g** [comment]
Alrighty. I'll ping the team to look into this.
